// Copyright 2016 Silent Circle, LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
package axolotl;

option optimize_for = LITE_RUNTIME;

message MessageEnvelope {
  optional string name           = 1;
  optional string scClientDevId  = 2;
  optional uint32 device_id      = 3;
  optional bytes  supplement     = 4;
  optional bytes  message        = 5;
  optional string msgId          = 6;
  optional bytes  recvIdHash     = 7;
  optional bytes  senderIdHash   = 8;
  optional string recvDeviceId   = 9;
  optional bytes  recvDevIdBin   = 10;
  optional string uid            = 11;
}
